### ISSUE TITLE
Error: Could not get remote files

### ISSUE THREAD

**Issue by virtualJonesie** [sakowicz/actual-ai#25]
When running the actual-ai Docker container, I get the following error:

```
> @sakowicz/actual-ai@1.2.1 dev-docker
> ../node_modules/.bin/nodemon --exec "npm run lint && node ./app.js"

[nodemon] 2.0.22
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `npm run lint && node ./app.js`

> @sakowicz/actual-ai@1.2.1 lint
> eslint --fix **/*.js

Application started
Starting classification process
An error occurred: Error: Could not get remote files
    at handlers.api/download-budget (/opt/node_app/node_modules/@actual-app/api/dist/app/bundle.api.js:29444:31)
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: Could not get remote files
    at handlers.api/download-budget (/opt/node_app/node_modules/@actual-app/api/dist/app/bundle.api.js:29444:31)

Node.js v18.20.4
[nodemon] app crashed - waiting for file changes before starting...

```

I am self-hosting the actual-server on my Synology NAS. As a result, I am using a self-signed SSL certificate. I believe the is the cause of my issue. I received the same error with actual-helpers, and the developer of that project was able to fix it by adding a config option to permit the actual-api to allow self-signed SSL certs.

https://github.com/psybers/actual-helpers/issues/2

These are the changes I see via GitHub, but I have asked the project owner for further details on the fix, and will provide updates as I get them.
```

Changed files for Self-Signed Certifiates

Dockerfile
Add:
# Allow self-signed SSL certs
ENV NODE_TLS_REJECT_UNAUTHORIZED=0


.env
Add:
# allow self-signed SSL certs
NODE_TLS_REJECT_UNAUTHORIZED=0
```

Thanks in advance for the help!

**Comment by sakowicz** [comment]
Could you try add that environment variable to the container?

Like this:
```
actual-ai:
    image: docker.io/sakowicz/actual-ai:latest
    restart: unless-stopped
    environment:
      - ACTUAL_SERVER_URL=http://actual_server:5006
      - ACTUAL_PASSWORD=your_actual_password
      - ACTUAL_BUDGET_ID=your_actual_budget_id # This is the ID from Settings → Show advanced settings → Sync ID
      - CLASSIFICATION_SCHEDULE_CRON=0 */4 * * * # How often to run classification.
      - CLASSIFY_ON_STARTUP=true # Whether to classify transactions on startup (don't wait for cron schedule)
      - SYNC_ACCOUNTS_BEFORE_CLASSIFY=false # Whether to sync accounts before classification
      - OPENAI_API_KEY=your_openai_api_key
#      - OPENAI_MODEL= # optional. required if you want to use a specific model, default is "gpt-3.5-turbo-instruct"
#      - OPENAI_BASE_URL= # optional. required if you don't want to use the OpenAI API but OpenAI compatible API
#      - ACTUAL_E2E_PASSWORD= # optional. required if you have E2E encryption
      - NODE_TLS_REJECT_UNAUTHORIZED=0
```

And let me know if it helped.

**Comment by virtualJonesie** [comment]
Same error

**Comment by virtualJonesie** [comment]
Update: My Docker skills need some work. Adding the environment variable to the docker-compose.yml did solve the specified problem. 

Now I'm getting a new error further along in the process. I'll troubleshoot and open a new issue, if needed.

Thanks!

**Comment by thibmaek** [comment]
For me this keeps occurring as well with the following `actualai.env` config mapped to the Docker container:

```
ACTUAL_SERVER_URL="http://172.17.0.1:5004"
ACTUAL_PASSWORD="..."
ACTUAL_BUDGET_ID="e3e2625b-..."
CLASSIFICATION_SCHEDULE_CRON="0 */4 * * *"
CLASSIFY_ON_STARTUP=true
SYNC_ACCOUNTS_BEFORE_CLASSIFY=false

LLM_PROVIDER=ollama
OPENAI_API_KEY=x
OLLAMA_BASE_URL=http://0.0.0.0:11434
OLLAMA_MODEL=llama3.2

NODE_TLS_REJECT_UNAUTHORIZED=0
```

And docker compose file:

```yaml
 actual_server:
    image: docker.io/actualbudget/actual-server
    container_name: actual_server
    env_file: actualbudget.env
    ports:
      - 5004:5006
    volumes:
      - ./data:/data
      - ./certs:/certs
    restart: unless-stopped

  actual_ai:
    image: docker.io/sakowicz/actual-ai
    container_name: actual_ai
    depends_on: [actual_server]
    env_file: actualai.env
    restart: unless-stopped
```

**Comment by sakowicz** [comment]
It's this the exact docker-compose file? Are these containers in the same network?

IMO, you should use `ACTUAL_SERVER_URL="http://actual_server:5004"` 

You can try to get into the actual_ai container and check via curl if the connection is OK.

**Comment by thibmaek** [comment]
Yep this is within the same `compose.yaml`. The IP I'm using here is the internal Docker IP which should always just route to the correct container in the engine.
Changing to the container name or service name from like you suggest has the same result. I can not `docker exec` into the container to open a shell because it keeps quitting/restarting/exiting 1
